src/meta/meta_ckpt.c

```diff
@@ -577,6 +720,52 @@ __wt_meta_ckptlist_to_meta(WT_SESSION_IMPL *session, WT_CKPT *ckptbase, WT_ITEM
     return (0);
 }
 
+/*
+ * __ckpt_blkmod_to_meta --
+ *     Add in any modification block string needed, including an empty one.
+ */
+static int
+__ckpt_blkmod_to_meta(WT_SESSION_IMPL *session, WT_ITEM *buf, WT_CKPT *ckpt)
+{
+    WT_BLOCK_MODS *blk;
+    WT_ITEM bitstring;
+    u_int i;
+    bool valid;
+
+    WT_CLEAR(bitstring);
+    valid = false;
+    for (i = 0, blk = &ckpt->backup_blocks[0]; i < WT_BLKINCR_MAX; ++i, ++blk)
+        if (F_ISSET(blk, WT_BLOCK_MODS_VALID))
+            valid = true;
+
+    /*
+     * If the existing block modifications are not valid, there is nothing to do.
+     */
+    if (!valid) {
+        WT_RET(__wt_buf_catfmt(session, buf, ",checkpoint_backup_info="));
+        return (0);
+    }
+
+    /*
+     * We have at least one valid modified block list.
+     */
+    WT_RET(__wt_buf_catfmt(session, buf, ",checkpoint_backup_info=("));
+    for (i = 0, blk = &ckpt->backup_blocks[0]; i < WT_BLKINCR_MAX; ++i, ++blk) {
+        if (!F_ISSET(blk, WT_BLOCK_MODS_VALID))
+            continue;
+        WT_RET(__wt_buf_catfmt(session, buf, "%s%s=(id=%" PRIu32 ",granularity=%" PRIu64
+                                             ",nbits=%" PRIu64 ",offset=%" PRIu64 ",blocks=",
+          i == 0 ? "" : ",", blk->id_str, i, blk->granularity, blk->nbits, blk->offset));
+        /* Get the number of bytes. */
+        WT_RET(__wt_raw_to_hex(session, blk->bitstring.mem, blk->nbits >> 3, &bitstring));
+        WT_RET(__wt_buf_catfmt(session, buf, "%.*s", (int)bitstring.size, (char *)bitstring.mem));
+        WT_RET(__wt_buf_catfmt(session, buf, ")"));
```

Comment:
You only need a single call to `__wt_buf_catfmt`, but I won't fight about it if you like the separation.

```
*** ../meta/meta_ckpt.c Tue Feb  4 20:12:51 2020
--- -   Tue Feb  4 20:37:14 2020 
***************  
*** 753,765 **** 
      for (i = 0, blk = &ckpt->backup_blocks[0]; i < WT_BLKINCR_MAX; ++i, ++blk) {
          if (!F_ISSET(blk, WT_BLOCK_MODS_VALID))
              continue;
-         WT_RET(__wt_buf_catfmt(session, buf, "%s%s=(id=%" PRIu32 ",granularity=%" PRIu64
-                                              ",nbits=%" PRIu64 ",offset=%" PRIu64 ",blocks=",
-           i == 0 ? "" : ",", blk->id_str, i, blk->granularity, blk->nbits, blk->offset));
-         /* Get the number of bytes. */
          WT_RET(__wt_raw_to_hex(session, blk->bitstring.mem, blk->nbits >> 3, &bitstring));
!         WT_RET(__wt_buf_catfmt(session, buf, "%.*s", (int)bitstring.size, (char *)bitstring.mem));
!         WT_RET(__wt_buf_catfmt(session, buf, ")"));
          __wt_buf_free(session, &bitstring);
      }   
      WT_RET(__wt_buf_catfmt(session, buf, ")"));
--- 753,764 ----
      for (i = 0, blk = &ckpt->backup_blocks[0]; i < WT_BLKINCR_MAX; ++i, ++blk) {
          if (!F_ISSET(blk, WT_BLOCK_MODS_VALID))
              continue;
          WT_RET(__wt_raw_to_hex(session, blk->bitstring.mem, blk->nbits >> 3, &bitstring));
!         WT_RET(__wt_buf_catfmt(session, buf, "%s%s=(id=%" PRIu32 ",granularity=%" PRIu64  
!                                              ",nbits=%" PRIu64 ",offset=%" PRIu64 ",blocks="
!                                            "%.*s)",
!           i == 0 ? "" : ",", blk->id_str, i, blk->granularity, blk->nbits, blk->offset,
!        (int)bitstring.size, (char *)bitstring.mem));
          __wt_buf_free(session, &bitstring);
      }   
      WT_RET(__wt_buf_catfmt(session, buf, ")"));
```

Reply:
Good catch. Fixed.